src/test/csf/BasicNetwork.h

```diff
@@ -240,12 +257,13 @@ template <class Function>
 inline void
 BasicNetwork<Peer>::send(Peer const& from, Peer const& to, Function&& f)
 {
+    using namespace std::chrono_literals;
     auto link = links_.edge(from,to);
     if(!link)
         return;
     time_point const sent = scheduler.now();
     scheduler.in(
-        link->delay,
+        std::max(0ns, link->delayGen()),
```

Comment:
Should we also enforce that messages are always sent in order? I believe the following is currently possible
```
t = 1   : send(a,b, mesg1);  with random delay of 50 seconds
t = 10 : send(a,b,mesg2); with random delay of 30 seconds
...
t= 40: mesg2 delivered
t= 51: mesg1 delievered
```
This might be an important degree of freedom to model, but I'm not sure whether reordering should be explicitly modeled instead.  Curious what you think.

Reply:
I think things make sense the way they are currently. In the future, if we want to model messages being reordered, then we can consider constraining messages to be sent in order. But then, in order to segment out the effects of the network on message delivery, we would almost need an intermediate entity to intercept all messages and apply some non-ideal policy (reordering, congestion, attacks, etc.). I think that would eventually be a useful feature, but I think it requires a lot more thought.

Reply:
I agree. I believe we use TCP for all the peer protocol layer, so probably not worth modeling reordering at this stage.  In that case, I think we still need to change the proposed implementation to prevent reordering.  I think we can just store a  `SimTime nextDelivery` in the link and then ensure that 
```
SimTime when = now() + std::max(0ns, link->delayGen());
when = std::max(when, link->nextDelivery + 1ns);
link->nextDelivery = when;
scheduler.at(when,...)
```